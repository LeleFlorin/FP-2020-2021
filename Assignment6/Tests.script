:l MiniLisp.hs

runParser ws "abc"
runParser ws " abc"

runParser (between (char '[') (char ']') (string "abc")) "[abc]xyz"
runParser (between (char '1') (char '1') (string "abc")) "1abc1xyz"
runParser (between (char '1') (char '1') (string "abc")) "[abc]xyz"
runParser (between (string "hey") (string "hey") (string "there")) "heythereheyxyz"

runParser ident "abc abc"
runParser ident " abc abc"
runParser ident "1abc abc"
runParser ident "abc1 abc"
runParser ident "empty? abc"

runParser (sepBy (char ',') ident) "a,b,c"
runParser (sepBy (char ',') ident) "abc"
runParser (sepBy (char ',') ident) "a,,b"
runParser (sepBy ws ident) "a b c d"

runParser lispAtom "abc"
runParser lispAtom "123"
runParser lispAtom "?123"
runParser lispAtom "(123)"

runParser lispList "(a b c)"
runParser lispList "(a b c("

runParser lisp "(a b c)"
runParser lisp "123"
runParser lisp "abc1"