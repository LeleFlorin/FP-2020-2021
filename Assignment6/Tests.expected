Success "" "abc"
Success " " "abc"

Success "abc" "xyz"
Success "abc" "xyz"
Error "Unexpected character '['"
Success "there" "xyz"

Success "abc" " abc"
Error "Unexpected character ' '"
Error "Unexpected character '1'"
Success "abc1" " abc"
Success "empty?" " abc"

Success ["a","b","c"] ""
Success ["abc"] ""
Success ["a"] ",,b"
Success ["a","b","c","d"] ""

Success (Symbol "abc") ""
Success (Number 123) ""
Error "Unexpected character '?'"
Error "Unexpected character '('"

Success [Atom (Symbol "a"),Atom (Symbol "b"),Atom (Symbol "c")] ""
Error "Unexpected character '('"

Success (List [Atom (Symbol "a"),Atom (Symbol "b"),Atom (Symbol "c")]) ""
Success (Atom (Number 123)) ""
Success (Atom (Symbol "abc1")) ""